mongodb/intro/crud-i/query-collection-3.md

```diff
@@ -0,0 +1,172 @@
+---
+author: Stefan-Stojanovic
+
+levels:
+  - beginner
+
+aspects:
+  - new
+  - workout
+
+type: normal
+
+category: how to
+
+---
+
+# Querying a collection using $and
+
+---
+## Content
+
+In the previous insight, we discussed how to query using the `$lt`, `$gt`, `$lte` and `$gte` operators.
+
+Now, we are going to discuss how to define a range between two values using a combination of the 4 previously explained operators together with the `$and` logical operator.
+
+The `$and` logical operator is used to join two or more expressions together to query data more precisely. When we query with the `$and` operator, no matter the number of expressions located inside, all expressions have to evaluate to true.
+
+Let's say we want to find all the pokémon whose `power` ranges between 250 and 350. For this, we would need to use 2 or more of the above operators together with the `$and` operator. If we would use them without `$and` it would not give a range between the two values, but rather all values lower than and greater than the specified ones.
```

Comment:
This is incorrect. The actual query would be collapsed and the last filter on `power` will be used, returning all pokemons with power less than `450`

So
```javascript
db.pokemon.find({
  power: { $gt: 250 },
  power: { $lt: 450 }
})
```

is same as 

```javascript
db.pokemon.find({
  power: { $lt: 450 }
})
```

Reply:
Looking at this insight, there is no need to use `$and` to define ranges. They could be simply defined like this:
```
db.pokemon.find({
  power: { $gt: 250, $lt: 450 }
});
```

Reply:
I've removed the `$and` concept from this insight and replaced it with the syntax above.

Reply:
The $and operator makes sure the range is between the 2 numbers

Reply:
As sourabh said, the 2 examples above give out the same output. However, if you add an $and operator you create a range between the 2 values

Reply:
This:
```javascript
db.pokemon.find(
{
    power:{$gt:250},
    power:{$lt:450}
})
```
and this:
```javascript
db.pokemon.find(
{
    power:{$lt:450}
})
give the same output(all documents with power Less than 450)

However this:
```
db.pokemon.find(
    {$and: 
        [
        {power: {$gte: 250}},
        {power: {$lte:350}}
        ]
    }
)
```
makes sure only the range between the 2 values is used and only pokemon with power between the 2 values is outputed

Reply:
I understand that, but I don't see why you would use the `$and` operator, when the same could be achieved without using it:
```
db.pokemon.find(
    {$and: 
        [
        {power: {$gte: 250}},
        {power: {$lte:350}}
        ]
    }
);
// is the same as this
db.pokemon.find({
  power: {$gte: 250, $lte: 250}
});
```